src/utils.js

```diff
@@ -437,8 +438,11 @@ const extractUrls = ({ string, urlRegExp = URL_NO_COMMAS_REGEX }) => {
  * @memberOf utils
  */
 const getRandomUserAgent = () => {
-    const index = getRandomInt(USER_AGENT_LIST.length);
-    return USER_AGENT_LIST[index];
+    if (userAgentList === null) {
+        const generateUserAgent = new UserAgents({ deviceCategory: 'desktop', userAgent: /^Mozilla/ });
+        userAgentList = Array(500).fill().map(() => generateUserAgent());
```

Comment:
This is better, except for one small thing - if there are less than 500 matching UAs, this array will be empty. IMHO it's better to use the custom filter function.

Reply:
I think we can live with top 50 at most. There's 83 in this list: https://techblog.willshouse.com/2012/01/03/most-common-user-agents/ and half of them has below 1% share.

The list is auto-updated.

Reply:
with this list we have 
```
Chrome / Chromium {72, 71, 70}
Firefox {65, 64}
Safari {12, 11}
Edge {18, 17, 16}
IE {11}
```
A custom filter function can be used on agent.family =[Chrome, Chromium, Firefox, Safari, Edge, IE] with the corresponding versions.

Reply:
If I take the values of the latest versions, there are 10103 results!
```
let userAgentList = null;
const getRandomUserAgent = () => {
    if (userAgentList === null) {
        let countUA = 0;
        const generateUserAgent = new UserAgents((data) => {
            if (data.deviceCategory !== 'desktop') return false;
            if (!data.userAgent.startsWith('Mozilla')) return false;

            const agent = useragent.parse(data.userAgent);
            switch (agent.family) {
            case 'Chrome':
            case 'Chromium':
                return (agent.major >= 72);
            case 'Firefox':
                return (agent.major >= 65);
            case 'Safari':
                return (agent.major >= 12);
            case 'Edge':
                return (agent.major >= 18);
            default:
                return false;
            }
        });

        // const generateUserAgent = new UserAgents({ deviceCategory: 'desktop', userAgent: /^Mozilla/ });
        // userAgentList = Array(50).fill().map(() => generateUserAgent());
    }

    return userAgentList[getRandomInt(userAgentList.length)].toString();
};
```
Should we take 50 or more randomly among the 10103 or the first {50,500}?

Reply:
IMHO we can just go with something like this:

```js
let userAgentList = null;
const getRandomUserAgent = () => {
    if (userAgentList === null) {
            let count = 0;
            userAgentList = new UserAgents((data) => {
                    if (data.deviceCategory !== 'desktop') return false;
                    if (!data.userAgent.startsWith('Mozilla')) return false;
                    if (++count > MAX_USER_AGENTS_COUNT) return false;
                    return true;
            });
    }
    return userAgentList.random().toString();
};
```

Reply:
Why not :)
This selects the `MAX_USER_AGENTS_COUNT` first user agents.
I'm not sure it selects `MAX_USER_AGENTS_COUNT` most-common user agents.

Reply:
I think it does, the UAs are sorted in the file based on the weight

Reply:
Ah OK. 
I couldn't find any documentation on the file description.
I commit what you propose. Thanks for our help.

Reply:
@jancurn you're right! I have received confirmation that the UAs are sorted by decreasing weight.